src/utils/Cypher.ts

```diff
@@ -26,19 +35,193 @@ export default class Cypher {
+    } else if (key === this.cryptoSetting?.key) {
+      algorithm = `${this.cryptoSetting.keySetting.blockCypher.toLowerCase()}-${this.cryptoSetting.keySetting.keySize}-${this.cryptoSetting.keySetting.operationMode.toLowerCase()}`;
+    }
+    return algorithm;
+  }
+
+  public static getCryptoKeySync(): string {
+    return this.cryptoSetting?.migrationDone ? this.cryptoSetting?.key : this.cryptoSetting?.formerKey;
+  }
+
+  public static async getCryptoKey(tenantID: string): Promise<string> {
+    if (!this.cryptoSetting) {
+      this.keySetting = await SettingStorage.getCryptoSettings(tenantID);
+      this.cryptoSetting = this.keySetting.crypto;
+    }
+    return this.cryptoSetting?.migrationDone ? this.cryptoSetting?.key : this.cryptoSetting?.formerKey;
+  }
+
+  public static async getMigrationDone(tenantID: string): Promise<{ migrationDone: boolean, migrationId: string }> {
+    this.keySetting = await SettingStorage.getCryptoSettings(tenantID);
+    this.cryptoSetting = this.keySetting.crypto;
+    return {
+      migrationDone: this.cryptoSetting.migrationDone,
+      migrationId: this.cryptoSetting.sensitiveDataMigrationId
+    };
+  }
+
+  public static async setMigrationDone(tenantID:string): Promise<void> {
+    this.cryptoSetting.migrationDone = true;
+    this.cryptoSetting.sensitiveDataMigrationId = null;
+    const keySettingToSave = {
+      id: this.keySetting.id,
+      identifier: TenantComponents.CRYPTO,
+      type: CryptoSettingsType.CRYPTO,
+      crypto: this.cryptoSetting
+    } as KeySettings;
+    await SettingStorage.saveCryptoSettings(tenantID, keySettingToSave);
+  }
+
+  public static async saveMigrationId(tenantID:string, migrationId: string): Promise<void> {
+    this.cryptoSetting.sensitiveDataMigrationId = migrationId;
+    const keySettingToSave = {
+      id: this.keySetting.id,
+      identifier: TenantComponents.CRYPTO,
+      type: CryptoSettingsType.CRYPTO,
+      crypto: this.cryptoSetting
+    } as KeySettings;
+    await SettingStorage.saveCryptoSettings(tenantID, keySettingToSave);
+  }
+
+  public static getSensitiveDatainSetting(setting: SettingDB): string[] {
+
+    const allSensitiveData: string[] = [];
+    for (const property of setting.sensitiveData) {
+      // Check that the property does exist otherwise skip to the next property
+      if (_.has(setting, property)) {
+        const value:string = _.get(setting, property);
+
+        // If the value is undefined, null or empty then do nothing and skip to the next property
+        if (value && value.length > 0) {
+          allSensitiveData[`${property}`] = value;
+        }
+      }
+    }
+    return allSensitiveData;
+
+  }
+
+  public static prepareSaveSensitiveDataMigration(setting: SettingDB,
+    valueBeforeDecrypt: string[], valueAfterDecrypt: string[], valueAfterEncrypt: string[]): SettingSensitiveData {
```

Comment:
I should have only one method in this class to migrate sensitive data of a tenant.